bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,291 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.salus.internal.rest.Device;
+import org.openhab.binding.salus.internal.rest.DeviceProperty;
+import org.openhab.binding.salus.internal.rest.GsonMapper;
+import org.openhab.binding.salus.internal.rest.JettyHttpClient;
+import org.openhab.binding.salus.internal.rest.SalusApi;
+import org.openhab.core.common.ThreadPoolManager;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public final class CloudBridgeHandler extends BaseBridgeHandler implements CloudApi {
+    private Logger logger = LoggerFactory.getLogger(CloudBridgeHandler.class.getName());
+    private final HttpClientFactory httpClientFactory;
+    @NonNullByDefault({})
+    private LoadingCache<String, SortedSet<DeviceProperty<?>>> devicePropertiesCache;
+    @NonNullByDefault({})
+    private String username;
+    @NonNullByDefault({})
+    private char[] password;
+    @NonNullByDefault({})
+    private String url;
+    private long refreshInterval;
+    private long propertiesRefreshInterval;
+    @Nullable
+    private SalusApi salusApi;
+    @Nullable
+    private ScheduledFuture<?> scheduledFuture;
+
+    public CloudBridgeHandler(Bridge bridge, HttpClientFactory httpClientFactory) {
+        super(bridge);
+        this.httpClientFactory = requireNonNull(httpClientFactory, "httpClientFactory");
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            internalInitialize();
+            updateStatus(ONLINE);
+        } catch (Exception ex) {
+            logger.error("Cannot start server!", ex);
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, "Cannot start server! " + ex.getMessage());
+        }
+    }
+
+    private void internalInitialize() {
+        loadConfigs();
+        var missingUsername = "".equals(username);
+        var missingPassword = password == null || password.length == 0;
+        var missingUrl = "".equals(url);
+        if (missingUsername || missingPassword || missingUrl) {
+            var sb = new StringBuilder();
+            sb.append("Missing configuration!\n");
+            sb.append(missingUsername ? "❌" : "✅").append(" username\n");
+            sb.append(missingPassword ? "❌" : "✅").append(" password\n");
+            sb.append(missingUrl ? "❌" : "✅").append(" url\n");
+            sb.append("Please check your configuration!\n");
+            logger.error("{}", sb);
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, sb.toString());
+            return;
+        }
```

Comment:
See earlier comment about config, This could then be changed to:
```suggestion
        config = getConfigAs(SalusConfiguration.class);
        
        // Validate configuration
        if (this.config.vin.isBlank()) {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Your error message");
            return;
        }
```